.github/workflows/py-unittests.yml

```diff
@@ -7,9 +7,15 @@ on:
   push:
     branches: [main]
 
+# If a new commit is pushed, cancel the jobs from previous commits.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
```

Comment:
This change is actually unrelated to the initial purpose of this PR. This change makes it so: if you push a commit, existing jobs get cancelled. I like this one for keeping runners available/ reducing usage. But it is unrelated to this PR in general, and can be removed if you want.

Reply:
I think this makes sense.